yaas/host.py

```diff
@@ -1,4 +1,41 @@
 # coding: utf-8
 
+from __future__ import absolute_import
+from __future__ import print_function
+
+import inspect
+import pprint
+import requests
+
+from . import config
+
+
 def command(parser, args):
-    pass
+    """ Interact with server agents. """
+    subparsers = parser.add_subparsers(dest='subcommand')
+    subcommands = {
+        'list': host_list
+        }
+    for name, fn in subcommands.items():
+        subparsers.add_parser(name, help=inspect.getdoc(fn))
+    subargs, extra = parser.parse_known_args(args)
+    subcommands[subargs.subcommand](
+        subparsers.choices[subargs.subcommand], extra
+        )
+
+
+def host_list(parser, args):
+    """ List all server agents. """
+    r = requests.get(
+        '{scheme}://{server}:{port}/api/v1/hosts'.format(
+            scheme=config.scheme,
+            server=config.server,
+            port=config.port
+            ),
+        **config.requests_opts()
+        )
+    if config.args.verbose:
+        pprint.pprint(r.json())
```

Comment:
This will get really annoying. I'm leaning more and more towards wrapping requests. That way we can do something like res.raise_on_error() in one place.